src/common/Gateway.sol

```diff
@@ -397,14 +388,21 @@ contract Gateway is Auth, IGateway, Recoverable {
     function endBatching() external auth {
         require(isBatching, NoBatched());
 
-        for (uint256 i; i < batchLocators.length; i++) {
-            BatchLocator memory locator = batchLocators[i];
-            _send(locator.centrifugeId, locator.poolId, outboundBatch[locator.centrifugeId][locator.poolId]);
-            delete outboundBatch[locator.centrifugeId][locator.poolId];
-            delete batchGasLimit[locator.centrifugeId][locator.poolId];
+        bytes32[] memory locators = TransientArrayLib.getBytes32(BATCH_LOCATORS_SLOT);
+        for (uint256 i; i < locators.length; i++) {
+            uint16 centrifugeId = uint16(bytes2(locators[i]));
+            PoolId poolId = PoolId.wrap(uint64(bytes8(locators[i] << 16)));
```

Comment:
I would move this and its counterpart method as internal methods that we can easily test, ensuring always behave identically. If you agree, I can do in the gateway-uts once we merge this.

Reply:
Good idea! Great if you can do it